docs/api/memory/link_time_optimization.md

```diff
@@ -0,0 +1,29 @@
+# Link Time Optimization
+
+Link Time Optimization (LTO) is a program memory usage optimization mechanism performed by compiler at link time. At compile time compiler creates special intermediate represention of all translation units and then optimize them as a single unit at link time resulting in much better optimization comparing to non-LTO builds.
+
+## Using LTO in Mbed OS
+
+In Mbed OS build system LTO is implemented as an optional profile extension placed in `tools\profiles\extensions\lto.json`. When enabled, build profile is amended with LTO flags. LTO perform heavy memory optimizations what breaks debugging, so it's not recommended to enable it in debug/develop profiles.
+
+To enable LTO retype `--profile` option with LTO file path `tools\profiles\extensions\lto.json`
+
+<span class="notes">**Note**: For profile extensions you have to put full path relative to Mbed OS root folder.</span>
```

Comment:
```suggestion
<span class="notes">**Note**: For profile extensions you have to put full path relative to the project's root folder.</span>
```
E.g. when building `mbed-cloud-client-example` I have to compile with `mbed compile -t GCC_ARM -m K64F --profile mbed-os/tools/profiles/release.json --profile mbed-os/tools/profiles/extensions/lto.json`. Notice that `mbed-os` is present in the path.

Reply:
fixed